File: backend/Dtos/response/DietaResponseDTO.cs
using backend.Dtos.response;

public record DietaResponseDTO(
    int Id_Dieta,
    string Nombre,
    string Descripcion,
    DateTime Fecha_Inicio,
    DateTime Fecha_Fin,
    string Estado,
    string PacienteNombre,
    List<DietaComidaResponseDTO> Comidas
);

File: frontend/src/types/dto/response/DietaResponseDTO.ts
import type { DietaComidaResponseDTO } from './DietaComidaResponseDTO';

export interface DietaResponseDTO {
    id_Dieta: number;
    nombre: string;
    descripcion: string;
    fecha_Inicio: string;
    fecha_Fin: string;
    activa: boolean;
    pacienteNombre: string;
    comidas: DietaComidaResponseDTO[];
}